### PR TITLE
Cursory View of History: Don't check outgoing asset transfers unless necessary

### LINKED ISSUE #2059: Only request outgoing transfers with `getAssetTransfers` when necessary.
This should remove nearly half our calls to the Alchemy transfers API.

### PR DESCRIPTION
**Pull request by mhluongo**
Alchemy only supports checking against a "to" or "from" address when calling `getAssetTransfers`... meaning we are often making two API calls per account tracking.

This PR skips checking outgoing asset transfers when possible in `ChainService`. It should reduce `getAssetTransfer` calls by around 45% every 15 minutes (after the initial history load).

# To test

- [ ] Drop a `console.log` in `background/lib/alchemy.ts#getAssetTransfers`. Thanks to this being a call over a websocket, I didn't see an obvious way to instrument in dev tools.
- [ ] Add a read-only account
- [ ] Set a 15 minute timer on your phone :grin: 
- [ ] At the end of the period, count the number of `getAssetTransfer` calls. In my testing against `main`, I saw 40 calls in 15 minutes... and on this branch, 20 calls in 15 minutes. Note that some of that reduction is from the last PR, and my numbers are so high because I have 4 networks turned on (versus two).

Closes #2059

### CODE REVIEW

**Comment by mhluongo** [comment]
I was planning to build up to sniping #2042 in this PR, but this is already large enough and should provide a ton of benefits already — this PR saves tens of requests per user per 15 minutes, that idea saves 1 request per new user. I'll circle back to #2042 once I've thought through the necessary refactor a bit more (or if anyone else wants to take a stab).

**Comment by 0xDaedalus** [comment]
> Thanks to this being a call over a websocket, I didn't see an obvious way to instrument in dev tools.

FYI you can monitor and filter on individual messages in the network tab:
Screen Shot 2022-08-29 at 10 30 53 AM

**Comment by mhluongo** [comment]
> FYI you can monitor and filter on individual messages in the network tab:

Wooooh, huge ergo improvement, ty :smile:

**Review by 0xDaedalus — APPROVED** [review]
Looks good to me - left one non-blocking comment

File: background/services/chain/index.ts [hunk 1]
```diff
@@ -1068,6 +1045,20 @@ export default class ChainService extends BaseService<Events> {
     )
   }
 
+  /**
+   * Check for any incoming asset transfers involving tracked accounts.
+   */
+  private async handleRecentIncomingAssetTransferAlarm(): Promise<void> {
+    const accountsToTrack = await this.db.getAccountsToTrack()
+
+    await Promise.allSettled(
```
[the remark this answers is not in the document]
Comment (mhluongo) [reply background/services/chain/index.ts:1054]: Agreed — this was just a copy-paste from what we were already doing, but at a minimum this needs some jitter between accounts. Ideally we wouldn't be fetching all of them either :eyes:

**Pull request merged**